lib/puppet/provider/service/windows.rb

```diff
@@ -127,7 +127,7 @@ def logonaccount_insync?(current)
     @normalized_logon_account ||= normalize_logonaccount
     @resource[:logonaccount] = @normalized_logon_account
 
-    insync = @resource[:logonaccount] == current
+    insync = @resource[:logonaccount].downcase == current.downcase
```

Comment:
Perhaps use [casecmp?](https://rubyapi.org/o/string#method-i-casecmp-3F) here? I think it's equivalent because https://rubyapi.org/o/string#method-i-casecmp says it's equal to downcasing both.
```suggestion
    insync = @resource[:logonaccount].casecmp?(current)
```

Reply:
Hmm. This assumes that Windows insensitive string matches perform case folding...... That seems like a reasonable assumption to make for now.

Reply:
Reading the docs it suggests downcase does the same